### PR TITLE
Add back changes for GitHookController support for emergencies

### PR DESCRIPTION
**Pull request by asheepapart**
Effectively, Revert "Revert "Merge pull request #23 from box/ashar-emergencies""
This reverts commit b62b2f22eaf6f60cd44f23181f7fa47a561b4e31.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @asheepapart has signed the CLA. Thanks for the pull request!

File: test/Bart/GitHook/GitHookControllerTest.php [hunk 1]
```diff
@@ -115,17 +116,40 @@ public function testProcessRevisionWithMultipleIncludedRefsAndRevs()
...
 	 * @param int $numValidRefs Number of valid refs in the standard input array that are actually in $validRefs
 	 */
-	private function runProcessRevisionTest(array $stdInArray, array $revList, array $validRefs, $numValidRefs)
+	private function runProcessRevisionTest(array $stdInArray, array $revList, array $validRefs, $numValidRefs, $emergency = false)
 	{
+
 		$numInputs = count($stdInArray);
 		$numRevs = count($revList);
 
+        if ($emergency) {
```
Comment (asheepapart) [inline test/Bart/GitHook/GitHookControllerTest.php]: You can type this as 

```php
$message = $emergency ? 'EMEREGENCY commit' : 'Commit message ignored';
```

File: src/Bart/GitHook/GitHookController.php [hunk 2]
```diff
@@ -156,10 +159,20 @@ private function processRevisions()
...
-	private function shouldSkip(Commit $commit)
+	private function shouldSkip(Commit $commit, GitHookConfig $gitHookConfig)
 	{
 		$message = $commit->message();
 
-		return preg_match('/^EMERGENCY/', $message) === 1;
+		$isEmergency = preg_match('/^EMERGENCY/', $message) === 1;
+
+		if ($isEmergency) {
+			$to = $gitHookConfig->getEmergencyNotificationEmailAddress();
```
Comment (asheepapart) [inline src/Bart/GitHook/GitHookController.php]: You're missing code to *not* send an email if these configs are empty.

**Comment by asheepapart** [comment]
Thanks for staying on top of this, @arizqi!

**Pull request merged**